dynamodb/src/main/java/org/apache/metamodel/dynamodb/DynamoDbTableCreationBuilder.java

```diff
@@ -0,0 +1,111 @@
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.TableStatus;
+
+class DynamoDbTableCreationBuilder extends AbstractTableCreationBuilder<DynamoDbUpdateCallback> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDbTableCreationBuilder.class);
+
+    public DynamoDbTableCreationBuilder(DynamoDbUpdateCallback updateCallback, Schema schema, String name) {
+        super(updateCallback, schema, name);
+    }
+
+    @Override
+    public Table execute() throws MetaModelException {
+        final MutableTable table = getTable();
+        final String tableName = table.getName();
+
+        final Collection<AttributeDefinition> attributes = new ArrayList<>();
+        final Collection<KeySchemaElement> keySchema = new ArrayList<>();
+        final Collection<GlobalSecondaryIndex> globalSecondaryIndices = new ArrayList<>();
+
+        final long readCapacity = Long.parseLong(System.getProperty(
+                DynamoDbDataContext.SYSTEM_PROPERTY_THROUGHPUT_READ_CAPACITY, "5"));
+        final long writeCapacity = Long.parseLong(System.getProperty(
+                DynamoDbDataContext.SYSTEM_PROPERTY_THROUGHPUT_WRITE_CAPACITY, "5"));
+        final ProvisionedThroughput provisionedThroughput = new ProvisionedThroughput(readCapacity, writeCapacity);
+
+        final Column[] columns = table.getColumns();
+        for (Column column : columns) {
+            if (column.isPrimaryKey()) {
+                final KeyType keyType = getKeyType(column.getRemarks());
+                keySchema.add(new KeySchemaElement(column.getName(), keyType));
+                attributes.add(new AttributeDefinition(column.getName(), DynamoDbUtils.toAttributeType(column
+                        .getType())));
+            }
+        }
+
+        final CreateTableRequest createTableRequest = new CreateTableRequest();
+        createTableRequest.setTableName(tableName);
+        createTableRequest.setAttributeDefinitions(attributes);
+        createTableRequest.setGlobalSecondaryIndexes(globalSecondaryIndices);
+        createTableRequest.setKeySchema(keySchema);
+        createTableRequest.setProvisionedThroughput(provisionedThroughput);
+
+        final AmazonDynamoDB client = getUpdateCallback().getDataContext().getDynamoDb();
+
+        final CreateTableResult createTableResult = client.createTable(createTableRequest);
+
+        // await the table creation to be "done".
+        {
+            String tableStatus = createTableResult.getTableDescription().getTableStatus();
+            while (TableStatus.CREATING.name().equals(tableStatus)) {
+                logger.debug("Waiting for table status to be ACTIVE. Currently: {}", tableStatus);
+                try {
+                    Thread.sleep(300);
+                } catch (InterruptedException e) {
+                }
```

Comment:
The best thing you can do if you can't stop executing, e.g. because of atomicity concerns is something like:
```java
            String tableStatus = createTableResult.getTableDescription().getTableStatus();
            boolean interrupted = false
            while (TableStatus.CREATING.name().equals(tableStatus)) {
                logger.debug("Waiting for table status to be ACTIVE. Currently: {}", tableStatus);
                try {
                    Thread.sleep(300);
                } catch (InterruptedException e) {
                    interrupted = true;
                }
                tableStatus = client.describeTable(tableName).getTable().getTableStatus();
            }
            if (interrupted) {
                 Thread.currentThread().interrupt();
            }
```

(Approach taken from https://www.ibm.com/developerworks/library/j-jtp05236/#N10186)